python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py

```diff
@@ -335,11 +334,239 @@ def write_parquet_file(context):
+        ),
+    )
+
+    ocsp_response_cache_filename: Optional[str] = Field(
+        default=None,
+        description=(
+            "URI for the OCSP response cache file.  By default, the OCSP response cache "
+            "file is created in the cache directory."
+        ),
+    )
+
+    validate_default_parameters: Optional[bool] = Field(
+        default=None,
+        description=(
+            "False by default. Raise an exception if either one of specified database, "
+            "schema or warehouse doesn't exists if True."
+        ),
+    )
+
+    paramstyle: Optional[str] = Field(
+        default=None,
+        description=(
+            "pyformat by default for client side binding. Specify qmark or numeric to "
+            "change bind variable formats for server side binding."
+        ),
+    )
+
+    timezone: Optional[str] = Field(
+        default=None,
+        description=(
+            "None by default, which honors the Snowflake parameter TIMEZONE. Set to a "
+            "valid time zone (e.g. America/Los_Angeles) to set the session time zone."
+        ),
+    )
+
+    connector: Optional[str] = Field(
+        default=None,
+        description=(
+            "Indicate alternative database connection engine. Permissible option is "
+            "'sqlalchemy' otherwise defaults to use the Snowflake Connector for Python."
+        ),
+        is_required=False,
+    )
+
+    cache_column_metadata: Optional[str] = Field(
+        default=None,
+        description=(
+            "Optional parameter when connector is set to sqlalchemy. Snowflake SQLAlchemy takes a"
+            " flag cache_column_metadata=True such that all of column metadata for all tables are"
+            ' "cached"'
+        ),
+    )
+
+    numpy: Optional[bool] = Field(
+        default=None,
+        description=(
+            "Optional parameter when connector is set to sqlalchemy. To enable fetching "
+            "NumPy data types, add numpy=True to the connection parameters."
+        ),
+    )
+
+    authenticator: Optional[str] = Field(
+        default=None,
+        description="Optional parameter to specify the authentication mechanism to use.",
+    )
+
+    @validator("paramstyle")
+    def validate_paramstyle(cls, v: Optional[str]) -> Optional[str]:
+        valid_config = ["pyformat", "qmark", "numeric"]
+        if v is not None and v not in valid_config:
+            raise ValueError(
+                "Snowflake Resource: 'paramstyle' configuration value must be one of:"
+                f" {','.join(valid_config)}."
+            )
+        return v
+
+    @validator("connector")
+    def validate_connector(cls, v: Optional[str]) -> Optional[str]:
+        if v is not None and v != "sqlalchemy":
+            raise ValueError(
+                "Snowflake Resource: 'connector' configuration value must be None or sqlalchemy."
+            )
+        return v
+
+    def get_client(self) -> SnowflakeConnection:
```

Comment:
my final question/concern is what should this method be called? Based on previous convos, it seems like it should be `get_connection` since it returns a `SnowflakeConnection` object. However, the snowflake library also has a `SnowflakeConnection` class (that we use in our SnowflakeConnection class). 

My concern is that `get_connection` might be confusing/misleading as users will think they are getting a snowflake library `SnowflakeConnection`, when in fact they are getting a dagster `SnowflakeConnection`

Another kind of annoying thing caused by our own `SnowflakeConnector` class is that using the snowflake resource is a bit different than other db resources. For example 

```
@asset 
def my_asset(snowflake: SnowflakeResource):
    snowflake.get_client() # Returns dagster_snowflake.SnowflakeConnection
    with snowflake.get_client().get_connection() as conn:
           # conn is a snowflake.SnowflakeConnection
          conn.execute("SELECT *")
    # but we also have this method
    snowflake.get_client().execute_query("SELECT *") 
```
as an example, BigQuery is much more bare bones
```
@asset 
def my_asset(bigquery: BigQueryResource):
     with bigquery.get_client() as client:
        # client is a bigquery.Client
       client.query("SELECT *")
```

cc @yuhan  @benpankow  @schrockn  for your thoughts

Reply:
Sounds like the core problem is pre-existing: That both we and snowflake have a `SnowflakeConnection` class. Being forced to write a `get_connection` method poke at the existing problem.

Does `dagster_snowflake.SnowflakeConnection` add sufficient value over `snowflake.SnowflakeConnection`?

* If not we could use this opportunity to divorce ourselves from it and instead just have `get_connection` return `snowflake.SnowflakeConnection`
* If it does, we can name it something new and deprecate the old class name by having it be a trivial subclass for backwards compat.

Reply:
Our SnowflakeConnector provides some helper methods (execute_query, execute_queries, and a method to load a parquet file into snowflake) and, but the most complicated/helpful thing it does is handle authentication. I know of one user who is using the `execute_queries` helper methods, but in my opinion it's no more challenging to use the snowflake connection object to execute some queries, and it's easy enough to switch to using the snowflake connector directly

I'm putting up a commit with a potential solution where we dont use the Dagster SnowflakeConnection anymore in the new resource. It's a bit rough but let me know what you think. it'll be easy to revert if it doesn't make sense as a path forward

Reply:
here's some code snippets outlining how to do the same things with the old resource and the new resource if the new resource just returns a snowflake SnowflakeConnection 

```
@asset(
    required_resource_keys={"snowflake"}
)
def run_query_old(context):
    snowflake = context.resources.snowflake
    snowflake.execute_query("SELECT 1", fetch_results=True, use_pandas_result=True)

@asset
def run_query_new(snowflake: SnowflakeResource):
    with snowflake.get_connection() as conn:
        conn.cursor().execute("SELECT 1").fetch_pandas_all()


@asset(
    required_resource_keys={"snowflake"}
)
def run_queries_old(context):
    snowflake = context.resources.snowflake
    results = snowflake.execute_queries(["SELECT 1", "SELECT 2", "SELECT 3"], fetch_results=True, use_pandas_result=True)

@asset
def run_queries_new(snowflake: SnowflakeResource):
    queries = ["SELECT 1", "SELECT 2", "SELECT 3"]
    results = []
    with snowflake.get_connection() as conn:
        for q in queries:
            results.append(conn.cursor().execute(q).fetch_pandas_all())
```